### ISSUE TITLE
`value_protection=True` silently destroys extreme values in every numeric binned column - causes data loss with no warning and no user override

### ISSUE THREAD

**Issue by AishaJaff** [dataxid/dataxid-python#96]
## Description

Every numeric column routed to `TABULAR_NUMERIC_BINNED` encoding has its 5-7 largest and 5-7 smallest values permanently removed from the learnable value range before training. This happens silently on every `synthesize()` call with no warning, no log message, and no way for the user to disable it. The clipping threshold is not fixed, it is randomly 5, 6, or 7 per column per run, so the extent of data loss also varies between runs on identical data.

The mechanism is in `_reduce_numeric()` (`_analyze.py`, lines 221–222). When `value_protection=True` (the hardcoded default), the function collects all per-row minimum and maximum values into sorted lists and then deliberately skips the true extremes by indexing into them with a random offset:

```python
reduced_min = reduced_min_n[noise_threshold(min_threshold=5)]  # 6th–8th smallest
reduced_max = reduced_max_n[noise_threshold(min_threshold=5)]  # 6th–8th largest
```

`noise_threshold()` (`_privacy.py`, lines 16–17) returns `5 + int(3 * np.random.uniform())` - an unseeded random integer in [5, 7]. The bin range for the entire column is then built from these clipped bounds, making any value outside the range unrepresentable in the model and therefore absent from all synthetic output.

This is not formal differential privacy. When `epsilon` is `None` (which it always is, since it cannot be set through `ModelConfig`), the code bypasses the `private_bounds()` path and uses this random heuristic instead - providing no DP guarantee while still destroying data. The parameter `value_protection=False` exists in `compute_stats()` (`_analyze.py`, line 502) and fully restores the original range, but it is absent from `ModelConfig` and `synthesize()`, leaving users with no mechanism to control this behaviour.

## Steps to Reproduce

```python
import pandas as pd
import numpy as np
import warnings
from dataxid.pipeline._analyze import compute_stats
from dataxid.pipeline._encode import encode_columns
from dataxid.pipeline._decode import decode_columns

np.random.seed(42)

# 190 regular employees + 5 executives ($150K–$2M)
salaries = np.concatenate([
    np.random.normal(60000, 10000, 190),
    np.array([150000, 500000, 800000, 1200000, 2000000])
])
df = pd.DataFrame({
    "salary": salaries,
    "age": np.random.randint(25, 65, 195).astype(float)
})

with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    stats = compute_stats(df, list(df.columns))  # value_protection=True by default

enc = encode_columns(df, list(df.columns), stats)
dec = decode_columns({k: v.tolist() for k, v in enc.items()}, list(df.columns), stats)
out = pd.to_numeric(dec["salary"], errors="coerce")

print(f"Input  max: ${df['salary'].max():>12,.0f} | executives (>$100K): {(df['salary'] > 100000).sum()}")
print(f"Output max: ${out.max():>12,.0f} | executives (>$100K): {(out > 100000).sum()}")
print(f"Bin range:  ${stats['salary']['bins'][0]:,.0f} to ${stats['salary']['bins'][-1]:,.0f}")
```

The clipping happens entirely in the local pipeline before any data reaches the server.

## Expected Behavior

Synthetic output preserves the full value range of the training data. Extreme values that are legitimate data points such as executive salaries, critical medical readings, peak financial figures should be representable in the output distribution. If value clipping is applied as a privacy measure, the SDK should emit a warning describing the extent of the loss and provide a parameter to disable it when the user accepts the trade-off.

## Actual Behavior

```
Input  max: $2,000,000 | executives (>$100K): 5
Output max: $   84,632 | executives (>$100K): 0
Bin range:  $43,925 to $84,632
```

95.8% of the maximum value is silently destroyed. All 5 executives earning over $100K vanish from the synthetic data. No exception is raised, no warning is emitted, and nothing in the logs indicates that any clipping occurred.

**The clipping amount is non-deterministic.** Running `compute_stats()` 10 times on identical data produces different upper bounds:

```
[84632, 87202, 84632, 87202, 81905, 81905, 81905, 81905, 81905, 81905]
# same dataset, 10 runs → $5,297 variation in the preserved maximum
```

**Live API confirmation** - tested via `dataxid.synthesize()` on the same dataset:

```
Input max:  $2,000,000
Output max: $75,218
Data loss:  96.2%
```

The server-side model only ever learns values within the clipped bin range, so it cannot generate values above the ceiling regardless of `n_samples`.

**The internal fix works but is inaccessible.** Passing `value_protection=False` directly to `compute_stats()` fully restores the range:

```python
stats_fixed = compute_stats(df, list(df.columns), value_protection=False)
# → output max: $2,000,000 | executives preserved: 7
```

This parameter is confirmed present in `compute_stats()`  but absent from all 16 fields of `ModelConfig` and from the `synthesize()` signature, making it unreachable through any public API.

## Configuration

```python
config = dataxid.ModelConfig()  
```

## Environment

- **dataxid version**: 0.2.0
- **Python version**: 3.12.13
- **OS**: Linux 6.6.113+ (Google Colab)

---

## Fix Suggestion

Two-part fix:

**Part 1 - Emit a `UserWarning` when clipping is severe.**

In `_reduce_numeric()` in `_analyze.py`, after lines 221–222, compare the clipped maximum against the true maximum and warn when the loss exceeds a meaningful threshold:

```python
# _analyze.py - after line 222
actual_max = reduced_max_n[0]
actual_min = reduced_min_n[0]
if (reduced_max < actual_max * 0.9) or (reduced_min > actual_min and actual_min > 0):
    import warnings
    warnings.warn(
        f"value_protection clipped this column's range from "
        f"[{actual_min:.4g}, {actual_max:.4g}] to "
        f"[{reduced_min:.4g}, {reduced_max:.4g}] "
        f"({(actual_max - reduced_max) / actual_max * 100:.1f}% loss at maximum). "
        f"Extreme values will be absent from synthetic output. "
        f"Pass value_protection=False to preserve the full range.",
        UserWarning,
        stacklevel=4,
    )
```

**Part 2 - Expose `value_protection` in `ModelConfig` and `synthesize()`.**

The parameter already exists in `compute_stats()` . It just needs to be surfaced to the user:

```python
# _config.py - add to ModelConfig dataclass
@dataclass
class ModelConfig:
    ...
    value_protection: bool = True
    # When True (default), clips 5–7 extreme values per column for privacy.
    # Set False to preserve the full value range including outliers.
```

```python
# _analyze.py - pass it through in the compute_stats() call
stats = compute_stats(df, features, value_protection=config["value_protection"])
```

This gives users a documented, supported way to opt out of clipping when they understand the trade-off for example, in financial or medical datasets where extreme values carry meaningful signal.

**Note:** Version 0.3.0 renamed the parameter to `protect_rare`. But the bug itself is identical.

**Comment by BurakUlver** [comment]
That's rare-value protection on numeric bounds, not silent corruption. Default is Privacy(protect_rare=True); turn it off if you need the full range:

```python
dataxid.ModelConfig(privacy=dataxid.Privacy(protect_rare=False))
